library/common/engine.cc

```diff
@@ -51,14 +53,23 @@ envoy_status_t Engine::run(const std::string config, const std::string log_level
     // When we improve synchronous failure handling and/or move to dynamic forwarding, we only need
     // to wait until the dispatcher is running (and can drain by enqueueing a drain callback on it,
     // as we did previously).
+    int64_t start_time = std::chrono::duration_cast<std::chrono::milliseconds>(
+                             std::chrono::system_clock::now().time_since_epoch())
+                             .count();
```

Comment:
Envoy lints against using system time - I think I'll actually revisit this and avoid doing the timing within the library and simply make the closure `void`